File: frontend/lib/api/appointments.ts
/**
 * Appointment API endpoints
 */

import { apiGet, apiPost, apiPut, apiDelete } from './client';

export interface Appointment {
  id: string;
  pet_id: string;
  clinic_id: string;
  owner_id: string;
  appointment_date: string;
  status: 'pending' | 'confirmed' | 'completed' | 'cancelled' | 'no_show';
  reason: string;
  notes?: string;
  diagnosis?: string;
  prescription?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateAppointmentRequest {
  pet_id: string;
  clinic_id: string;
  appointment_date: string;
  reason: string;
  notes?: string;
}

export interface UpdateAppointmentRequest {
  status?: 'pending' | 'confirmed' | 'completed' | 'cancelled' | 'no_show';
  diagnosis?: string;
  notes?: string;
  prescription?: string;
}

/**
 * Get all appointments for current user (owner)
 */
export async function getMyAppointments() {
  return apiGet<Appointment[]>('/appointments/my');
}

/**
 * Get single appointment
 */
export async function getAppointment(appointmentId: string) {
  return apiGet<Appointment>(`/appointments/${appointmentId}`);
}

/**
 * Create new appointment
 */
export async function createAppointment(data: CreateAppointmentRequest) {
  return apiPost<Appointment>('/appointments', data);
}

/**
 * Update appointment (clinic only)
 */
export async function updateAppointment(
  appointmentId: string,
  data: UpdateAppointmentRequest
) {
  return apiPut<Appointment>(`/appointments/${appointmentId}`, data);
}

/**
 * Cancel appointment (owner only)
 */
export async function cancelAppointment(appointmentId: string) {
  return apiDelete<void>(`/appointments/${appointmentId}`);
}

/**
 * Get clinic appointments
 */
export async function getClinicAppointments(clinicId: string) {
  return apiGet<Appointment[]>(`/appointments/clinic?clinic_id=${clinicId}`);
}

File: backend/routers/appointments.py
"""
Router for appointment endpoints
"""
from fastapi import APIRouter, Depends, status
from auth import get_current_user, AuthUser
from services import AppointmentService
from schemas import AppointmentCreate, AppointmentUpdate, AppointmentResponse
from typing import List

router = APIRouter(prefix="/api/appointments", tags=["appointments"])


@router.post("", response_model=AppointmentResponse, status_code=status.HTTP_201_CREATED)
async def create_appointment(
    appointment: AppointmentCreate,
    current_user: AuthUser = Depends(get_current_user)
):
    """
    Create a new appointment (owner only)
    
    - Owner creates appointment for their pet at a clinic
    - Returns created appointment with ID
    """
    return await AppointmentService.create_appointment(current_user.user_id, appointment)


@router.get("/my", response_model=List[AppointmentResponse])
async def get_my_appointments(
    current_user: AuthUser = Depends(get_current_user)
):
    """
    Get all appointments for current user (owner)
    
    - Returns appointments sorted by date
    - Only shows appointments owned by current user
    """
    return await AppointmentService.get_user_appointments(current_user.user_id)


@router.get("/clinic", response_model=List[AppointmentResponse])
async def get_clinic_appointments(
    clinic_id: str,
    current_user: AuthUser = Depends(get_current_user)
):
    """
    Get all appointments for a clinic (clinic staff only)
    
    - Returns appointments sorted by date
    - Clinic must verify ownership (todo: add clinic verification)
    """
    return await AppointmentService.get_clinic_appointments(clinic_id)


@router.get("/{appointment_id}", response_model=AppointmentResponse)
async def get_appointment(
    appointment_id: str,
    current_user: AuthUser = Depends(get_current_user)
):
    """
    Get single appointment details
    
    - Owner or clinic staff can view
    - Returns full appointment details
    """
    return await AppointmentService.get_appointment(appointment_id)


@router.put("/{appointment_id}", response_model=AppointmentResponse)
async def update_appointment(
    appointment_id: str,
    update_data: AppointmentUpdate,
    current_user: AuthUser = Depends(get_current_user)
):
    """
    Update appointment (clinic staff only)
    
    - Add diagnosis, notes, prescription
    - Update appointment status
    - Only clinic owner can update
    """
    # Note: Need to verify clinic ownership
    # For now, just pass clinic_id as current_user (needs refactor for multi-clinic support)
    return await AppointmentService.update_appointment(
        appointment_id,
        current_user.user_id,
        update_data
    )


@router.delete("/{appointment_id}", status_code=status.HTTP_204_NO_CONTENT)
async def cancel_appointment(
    appointment_id: str,
    current_user: AuthUser = Depends(get_current_user)
):
    """
    Cancel appointment (owner only)
    
    - Sets appointment status to 'cancelled'
    - Only owner can cancel
    """
    await AppointmentService.cancel_appointment(appointment_id, current_user.user_id)
    return None
